source/agora/common/crypto/ECC.d

```diff
@@ -143,14 +143,28 @@ public struct Scalar
         return ret;
     }
 
+    public bool isValid() () nothrow @nogc @safe
+    {
+        const auto ED25519_L =  BitBlob!256("0x1000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed");
+        const auto ZERO =       BitBlob!256("0x0000000000000000000000000000000000000000000000000000000000000000");
+        return this.data > ZERO && this.data < ED25519_L;
+    }
+
     /// Return the point corresponding to this scalar multiplied by the generator
-    public Point toPoint () const nothrow @nogc @trusted
-    out (val) { assert(crypto_core_ed25519_is_valid_point(val.data[].ptr)); }
-    do {
-        Point ret = void;
-        if (crypto_scalarmult_ed25519_base_noclamp(ret.data[].ptr, this.data[].ptr))
+    public Point toPoint () const nothrow @nogc @trusted {
+    Point ret = void;
+    if (crypto_scalarmult_ed25519_base_noclamp(ret.data[].ptr, this.data[].ptr)!=0) 
+    {
+        debug { import std.stdio : writefln; try { writefln("Scalar is not valid: %s", this.data.toString); } catch (Exception) {} }
```

Comment:
These debug `writef`s are rather unusual. I'm not sure it's a good idea to spam these everywhere unless this code segment triggers often (and in reality it should never trigger).

writef should be reserved for debugging in a local branch.

Reply:
Yes I wasn't sure about keeping the debug statements. Now removed and the file is updated based on the style guide.